### PR TITLE
Add defaults to enable passing no arguments when creating a `pygame.Font()`

### PR DESCRIPTION
**Pull request by PurityLake**
#1672 

This is a suggested solution to the Font class. It simply makes the font name and size arguments options with default values of "freesansbold.ttf" and 12 respectively.

This wouldn't be a breaking change as the default usual behavior still works, it just made the args optional instead of mandatory.

### CODE REVIEW

File: src_c/font.c [hunk 1]
```diff
@@ -72,6 +72,8 @@ static const char resourcefunc_name[] = "getResource";
 #endif
 static const char font_defaultname[] = "freesansbold.ttf";
 
+static const int font_defaultsize = 12;
```
Comment (MyreMylar) [inline src_c/font.c:75]: Should we make this default size a bit higher? The pygame default font is tiny at 12 (it comes out only about 6 pixels high for some reason):

image

This is it at 24 instead:
image

On the other hand, other fonts will be large at 24

**Review by MyreMylar — COMMENTED** [review]
It seems like the first argument needs to nail down what it is called for this to work with keyword arguments

In the code it is called "font", in the typehints it is called "name", in the docs it is called "file_path".

**Comment by PurityLake** [comment]
I will sort that out this evening

File: src_c/font.c (hunk 1 above)
Comment (PurityLake) [reply src_c/font.c:75]: That is a good question, maybe 16 splitting the different between 12 and 24?

File: docs/reST/ref/font.rst [hunk 2]
```diff
@@ -170,7 +171,8 @@ solves no longer exists, it will likely be removed in the future.
    the height of the font in pixels. If the filename is ``None`` the pygame
    default font will be loaded. If a font cannot be loaded from the arguments
    given an exception will be raised. Once the font is created the size cannot
-   be changed.
+   be changed. If no arguments are given then the default font will be used and
```
Comment (oddbookworm) [inline docs/reST/ref/font.rst:174]: Maybe add a note in the docs that the parameter name changed in 2.1.4?

**Review by oddbookworm — CHANGES_REQUESTED** [review]
As Myre commented, the default font is super tiny at the default size
```
import pygame

pygame.init()

screen = pygame.display.set_mode((500, 500))

default_font = pygame.font.Font()
text1 = default_font.render("Hello World!", True, "red")

freesans_font = pygame.font.Font("freesansbold.ttf")
text2 = freesans_font.render("Hello World!", True, "red")

running = True
while running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
            
    screen.fill("black")
    screen.blit(text1, (0, 0))
    screen.blit(text2, (0, 250))
    pygame.display.flip()
```
image

File: docs/reST/ref/font.rst (hunk 2 above)
Comment (PurityLake) [reply docs/reST/ref/font.rst]: I have added that detail now.

**Comment by PurityLake** [comment]
I have updated the font size to 20, I feel 20 is quite big enough for it's purposes

File: buildconfig/stubs/pygame/font.pyi [hunk 3]
```diff
@@ -32,7 +32,7 @@ class Font:
     italic: bool
     underline: bool
     strikethrough: bool
-    def __init__(self, name: Optional[FileArg], size: int) -> None: ...
+    def __init__(self, filename: Optional[FileArg], size: int) -> None: ...
```
Comment (MyreMylar) [inline buildconfig/stubs/pygame/font.pyi:35]:
```suggestion
    def __init__(self, filename: Optional[FileArg] = None, size: int = None) -> None: ...
```

Comment (MyreMylar) [reply buildconfig/stubs/pygame/font.pyi:35]: At the minute the type hints don't work because they are missing the default values.

**Review by MyreMylar — CHANGES_REQUESTED** [review]
Other than the type hints, this seems good.

Fix that and I will approve 👍

File: buildconfig/stubs/pygame/font.pyi [hunk 4]
```diff
@@ -32,7 +32,7 @@ class Font:
     italic: bool
     underline: bool
     strikethrough: bool
-    def __init__(self, name: Optional[FileArg], size: int) -> None: ...
+    def __init__(self, filename: Optional[FileArg] = None, size: int = None) -> None: ...
```
Comment (MyreMylar) [inline buildconfig/stubs/pygame/font.pyi:35]:
```suggestion
    def __init__(self, filename: Optional[FileArg] = None, size: int = 20) -> None: ...
```

File: src_c/doc/font_doc.h [hunk 5]
```diff
@@ -8,7 +8,7 @@
 #define DOC_PYGAMEFONTGETFONTS "get_fonts() -> list of strings\nget all available fonts"
 #define DOC_PYGAMEFONTMATCHFONT "match_font(name, bold=False, italic=False) -> path\nfind a specific font on the system"
 #define DOC_PYGAMEFONTSYSFONT "SysFont(name, size, bold=False, italic=False) -> Font\ncreate a Font object from the system fonts"
-#define DOC_PYGAMEFONTFONT "Font(filename, size) -> Font\nFont(pathlib.Path, size) -> Font\nFont(object, size) -> Font\ncreate a new Font object from a file"
+#define DOC_PYGAMEFONTFONT "Font(filename=None, size=12) -> Font\nFont(filename, size) -> Font\nFont(pathlib.Path, size) -> Font\nFont(object, size) -> Font\ncreate a new Font object from a file"
```
Comment (MyreMylar) [inline src_c/doc/font_doc.h:11]:
```suggestion
#define DOC_PYGAMEFONTFONT "Font(filename=None, size=20) -> Font\nFont(filename, size) -> Font\nFont(pathlib.Path, size) -> Font\nFont(object, size) -> Font\ncreate a new Font object from a file"
```

**Review by MyreMylar — COMMENTED** [review]
Whoops looks like my last suggestion had the wrong default value for size, but upon reviewing I noticed it was still set to 12 all over the shop as well.

**Review by MyreMylar — APPROVED** [review]

File: src_c/doc/font_doc.h [hunk 6]
```diff
@@ -73,6 +73,7 @@ pygame.font.SysFont
 create a Font object from the system fonts
 
 pygame.font.Font
+ Font(filename=None, size=12) -> Font
```
Comment (MyreMylar) [inline src_c/doc/font_doc.h:76]:
```suggestion
 Font(filename=None, size=20) -> Font
```

File: src_py/ftfont.py [hunk 7]
```diff
@@ -34,7 +34,7 @@ class Font(_Font):
     __unull = "\x00"
     __bnull = b"\x00"
 
-    def __init__(self, file, size=-1):
+    def __init__(self, file=None, size=-1):
```
Comment (ankith26) [inline src_py/ftfont.py:37]: shouldn't this be `size=20` too?

Comment (PurityLake) [reply src_py/ftfont.py:37]: Yes it should, sorry, didn't think to change that

Comment (MyreMylar) [inline src_py/ftfont.py:37]:
```suggestion
    def __init__(self, file=None, size=20):
```

**Review by oddbookworm — APPROVED** [review]
LGTM!

**Pull request merged**